vm.c

```diff
@@ -0,0 +1,22 @@
+#include "vm.h"
+#include "libkern.h"
+
+/* This is the pointer to the extended end of kernel's image. See
+   vm_early_alloc for details. */
+char* vm_kernel_image_end;
+
+void vm_init(){
+    /* The end of the kernel's .bss section. Provided by the linker. */
+    extern char __ebss[];
+
+    vm_kernel_image_end = __ebss;
```

Comment:
I believe I can't! `__ebss` is a linker symbol. It does not exist in the memory, there is no address under which it's value would be stored. It makes no sense assigning values to it on runtime, as it is not available there. It makes sense, however, to dereference it, since it points to some address, but that's not a good place to store values, as it's the end of the `.bss` section indeed, and it will be used for other purposes. Copying the value of `__ebss` to a variable makes sense too, as the linker places a constant under `__ebss`:

<pre>
(gdb) disas km_init 
Dump of assembler code for function km_init:
   0xbfc00c98 <+0>:	lui	v0,0xa000
   0xbfc00c9c <+4>:	addiu	v0,v0,144
   0xbfc00ca0 <+8>:	jr	ra
   0xbfc00ca4 <+12>:	sw	v0,40(gp)
End of assembler dump.
</pre>